dashboard/app/models/levels/blockly.rb

```diff
@@ -562,4 +563,17 @@ def shared_functions
       SharedBlocklyFunction.where(level_type: type).map(&:to_xml_fragment)
     end.join
   end
+
+  # Display translated custom block text
+  def localized_shared_blocks(level_objects)
+    return level_objects if level_objects.blank?
+    level_objects.each do |level_object|
+      next if level_object.blank?
+      block_text = level_object[:config]["blockText"]
+      if block_text.present?
+        level_object[:config]["blockText"] = I18n.t("data.blocks.#{level_object[:name]}.text")
```

Comment:
@Hamms do we need to worry that we're modifying the existing object here, vs. returning a copy?  Not sure where this object is re-used, or if we get a fresh copy with each ActiveRecord query.

Reply:
I believe this object comes from `Block.block_options`: https://github.com/code-dot-org/code-dot-org/blob/fb8f7caa1249ec79c862f30513b4b883c9104f87/dashboard/app/models/block.rb#L45-L53

And is therefore created anew on each query. That being said, we should probably still be returning a copy here rather than modifying the original since that fits the pattern of how the rest of these localization methods works.

Reply:
@nkiruka I think this PR will be set to merge after addressing this last bit of feedback!